AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/StorageEngineAdapter+SQLite.swift

```diff
@@ -126,21 +125,20 @@ final class SQLiteStorageEngineAdapter: StorageEngineAdapter {
         }
     }
 
+    // MARK: - Save
     func save<M: Model>(_ model: M, condition: QueryPredicate? = nil, completion: @escaping DataStoreCallback<M>) {
          save(model, modelSchema: model.schema, condition: condition, completion: completion)
      }
 
-    func save<M: Model>(_ model: M,
-                        modelSchema: ModelSchema,
-                        condition: QueryPredicate? = nil,
-                        completion: DataStoreCallback<M>) {
+    func save<M: Model>(_ model: M, modelSchema: ModelSchema, condition: QueryPredicate? = nil, completion: DataStoreCallback<M>) {
         guard let connection = connection else {
             completion(.failure(DataStoreError.nilSQLiteConnection()))
             return
         }
         do {
             let modelType = type(of: model)
-            let modelExists = try exists(modelSchema, withId: model.id)
+            let modelIdentifier = model.identifier(schema: modelSchema)
+            let modelExists = try exists(modelSchema, withIdentifier: modelIdentifier)
```

Comment:
what happens when an existing model like `ModelCustomPkDefined` with custom primary key defined has its `dob` field modified? would this code treat it as a new saved object? is that expected behavior?

Reply:
yeah, the current behaviour is to have primary keys immutable. Updated the test models.